src/test/java/org/broadinstitute/hellbender/utils/pairhmm/VectorPairHMMUnitTest.java

```diff
@@ -0,0 +1,358 @@
+package org.broadinstitute.hellbender.utils.pairhmm;
+
+import com.google.common.base.Strings;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix;
+import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+public final class VectorPairHMMUnitTest extends BaseTest {
+    private final static boolean ALLOW_READS_LONGER_THAN_HAPLOTYPE = true;
+    private final static boolean DEBUG = false;
+    final static boolean EXTENSIVE_TESTING = true;
+    final N2MemoryPairHMM exactHMM = new Log10PairHMM(true); // the log truth implementation
+    final N2MemoryPairHMM originalHMM = new Log10PairHMM(false); // the reference implementation
+    final N2MemoryPairHMM loglessHMM = new VectorLoglessPairHMM();
+
+    @BeforeClass
```

Comment:
Could you add a check that avx is supported in the before class, and then throw a skip exception if it isn't?